spec/controllers/posts_controller_spec.rb

```diff
@@ -1415,12 +1589,12 @@
         expect(templateless.plucked_characters).to eq([[char1.id, char1.name]])
 
         # tags
-        expect(assigns(:post).settings.size).to eq(3)
-        expect(assigns(:post).content_warnings.size).to eq(3)
-        expect(assigns(:post).labels.size).to eq(3)
-        expect(assigns(:post).settings.map(&:id_for_select)).to match_array([setting1.id, setting2.id, '_other'])
-        expect(assigns(:post).content_warnings.map(&:id_for_select)).to match_array([warning1.id, warning2.id, '_other'])
-        expect(assigns(:post).labels.map(&:id_for_select)).to match_array([label1.id, label2.id, '_other'])
+        expect(assigns(:post).settings.size).to eq(0)
+        expect(assigns(:post).content_warnings.size).to eq(0)
+        expect(assigns(:post).labels.size).to eq(0)
+        expect(assigns(:settings).map(&:id_for_select)).to match_array([setting1.id, setting2.id, '_other'])
+        expect(assigns(:content_warnings).map(&:id_for_select)).to match_array([warning1.id, warning2.id, '_other'])
+        expect(assigns(:labels).map(&:id_for_select)).to match_array([label1.id, label2.id, '_other'])
```

Comment:
Yeah, I think this is the same as the above comment.

Reply:
For posterity: yes. This is now stored exclusively in an assigns rather than being pulled from the object, because it shouldn't change on the object _unless we expect it to have changed on the object_ and that is kind of explicitly not what preview is doing, so, assigns.